BDArmory/Control/MissileFire.cs

```diff
@@ -7102,8 +7204,10 @@ void GuardMode()
                                         if (BDArmorySettings.DEBUG_MISSILES) Debug.Log("[BDArmory.MissileFire] target behind terrain");
                                     }
                                 }
-                                MissileLaunchParams dlz = MissileLaunchParams.GetDynamicLaunchParams(CurrentMissile, guardTarget.Velocity(), guardTarget.CoM, -1, (CurrentMissile.TargetingMode == MissileBase.TargetingModes.Laser
-                                    && BDATargetManager.ActiveLasers.Count <= 0 || CurrentMissile.TargetingMode == MissileBase.TargetingModes.Radar && !_radarsEnabled && !CurrentMissile.radarLOAL));
+                                MissileLaunchParams dlz = MissileLaunchParams.GetDynamicLaunchParams(CurrentMissile, guardTarget.Velocity(), guardTarget.CoM, -1,
+                                    (CurrentMissile.TargetingMode == MissileBase.TargetingModes.Laser && BDATargetManager.ActiveLasers.Count <= 0 ||
+                                    CurrentMissile.TargetingMode == MissileBase.TargetingModes.Radar && !(_radarsEnabled || _sonarsEnabled) && !CurrentMissile.radarLOAL ||
+                                    CurrentMissile.TargetingMode == MissileBase.TargetingModes.Inertial && !(_radarsEnabled || _sonarsEnabled || _irstsEnabled)));
 
```

Comment:
might make sense to make a new bool and a switch(targetingMode){} segment above this that toggles the bool as needed to keep this manageable

Reply:
Using a switch expression, the `unguidedGuidedMissile` parameter here could be written more cleanly as
```csharp
CurrentMissile.TargetingMode switch
{
    MissileBase.TargetingModes.Laser => BDATargetManager.ActiveLasers.Count <= 0,
    MissileBase.TargetingModes.Radar => !(_radarsEnabled || _sonarsEnabled) && !CurrentMissile.radarLOAL,
    MissileBase.TargetingModes.Inertial => !(_radarsEnabled || _sonarsEnabled || _irstsEnabled),
    _ => false
}
```
Similarly for lines 6503—6505.